src/main/kotlin/uk/gov/dwp/dataworks/services/AuthenticationService.kt

```diff
@@ -60,7 +60,8 @@ class AuthenticationService {
      * Helper method to extract the Cognito username from a JWT Payload.
      */
     fun cognitoUsernameFromJwt(jwt: DecodedJWT): String {
-        val tokenJson = ObjectMapper().readTree(jwt.payload)
-        return tokenJson.get("cognito:username").asText()
+        return jwt.getClaim("cognito:username")?.asString()?:
+        jwt.getClaim("userName")?.asString()?:
+        throw IllegalArgumentException("No username found in JWT token")
```

Comment:
The formatting is hard to read on this, could you add newlines before the elvis operators instead of after?
Also we don't need the null checks before the `.asString()` as the methods are already null safe

Reply:
I didn't realise that it was null safe, handy! I have made the changes and pushed.